test/tools/PSxUnit/CommandLine.cs

```diff
@@ -0,0 +1,362 @@
+                }
+                // BEGIN: Special command line switches for dotnet <=> Visual Studio integration
+                else if (optionName == "test" || optionName == "-test")
+                {
+                    if (option.Value == null)
+                        throw new ArgumentException("missing argument for --test");
+
+                    DesignTimeTestUniqueNames.Add(option.Value);
+                }
+                else if (optionName == "list" || optionName == "-list")
+                {
+                    GuardNoOptionValue(option);
+                    List = true;
+                }
+                else if (optionName == "designtime" || optionName == "-designtime")
+                {
+                    GuardNoOptionValue(option);
+                    DesignTime = true;
+                }
+                else if (optionName == "port" || optionName == "-port")
+                {
+                    if (option.Value == null)
+                    {
+                        throw new ArgumentException("missing argument for -port");
+                    }
+
+                    int port;
+                    if (!int.TryParse(option.Value, out port) || port < 0)
+                    {
+                        throw new ArgumentException("incorrect argument value for -port (must be a positive number)");
+                    }
+
+                    Port = port;
+                }
+                else if (optionName == "wait-command" || optionName == "-wait-command")
+                {
+                    GuardNoOptionValue(option);
+                    WaitCommand = true;
+                }
+                // END: Special command line switches for dotnet <=> Visual Studio integration
+                else
+                {
+                    // Might be a reporter...
+                    var reporter = reporters.FirstOrDefault(r => string.Equals(r.RunnerSwitch, optionName, StringComparison.OrdinalIgnoreCase));
+                    if (reporter != null)
+                    {
+                        GuardNoOptionValue(option);
+                        if (Reporter != null)
+                            throw new ArgumentException("only one reporter is allowed");
+
+                        Reporter = reporter;
+                    }
+                    // ...or an result output file
+                    else
+                    {
+                        if (!TransformFactory.AvailableTransforms.Any(t => t.CommandLine.Equals(optionName, StringComparison.OrdinalIgnoreCase)))
+                            throw new ArgumentException($"unknown option: {option.Key}");
+
+                        if (option.Value == null)
+                            throw new ArgumentException(string.Format("missing filename for {0}", option.Key));
+
+                        project.Output.Add(optionName, option.Value);
+                    }
+                }
+            }
```

Comment:
Please break this into smaller functions. Some code might be reusable and it will become much more readable.